llvm/test/CodeGen/RISCV/rvv/vpbinops-scalarization.ll

```diff
@@ -0,0 +1,896 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
```

Comment:
InstCombine tests should be in test/Transforms/InstCombine/RISCV and shouldn't run llc.

Reply:
I'd like to check that when this InstCombine happens, we get the desired RISCV scalar code. Is there a way to do that without running opt followed up by llc? I have seen other tests run both before.

Reply:
It's generally encouraged to run both like that. There are definitely some CodeGen tests that do it, but not many.

Do you have a particular concern?

Scalarizing i64 vectors on rv32 will require 2 scalar instructions, possibly more depending on the operation.

i8/i16 shifts, div, and rem will be more instructions to sign extend or zero extend the inputs.

I think we can have f16 vector without having scalar f16 arithmetic so that would generate more code to convert to f32 and back. Maybe the backend should use VL=1 vectors in that case?

Reply:
Ok. I've moved it into InstCombine tests. It no longer needs to be RISCV specific and I reflected that by not placing it in InstCombine/RISCV tests.

Reply:
> I think we can have f16 vector without having scalar f16 arithmetic

I think that is possible too.

> that would generate more code to convert to f32 and back. Maybe the backend should use VL=1 vectors in that case?

Is it possible to query what the backend has from InstCombine so that we can decide whether or not we'd like to do this optimization?